packages/astro/src/@types/astro.ts

```diff
@@ -82,17 +82,20 @@ export interface TransitionAnimationPair {
 	new: TransitionAnimation | TransitionAnimation[];
 }
 
-export interface TransitionDirectionalAnimations {
+export interface TransitionStrictDirectionalAnimations {
 	forwards: TransitionAnimationPair;
 	backwards: TransitionAnimationPair;
 }
 
+export type TransitionFreeDirectionalAnimations = Record<string, TransitionAnimationPair>;
+
 export type TransitionAnimationValue =
 	| 'initial'
 	| 'slide'
 	| 'fade'
 	| 'none'
-	| TransitionDirectionalAnimations;
```

Comment:
There may be a need for discussion here: Currently, animations can only define "forward" and "backward", which are rendered as "" and "back". If we want to be able to support more different navigation directions on a page, we need more names. I've called the old behavior "strict" and defined a "free" variant where you can define animations with any name that will make it into the stylesheet without being changed or deleted.

Reply:
What is the reason for the change? Does it relate to the events at all?

I'm not sure what other directions there could be in the future, do you have some in mind?

Reply:
Is it about the fish demo? If so, i don't think that should be part of our API. From out perspective, there's a forward direction (click links, submit forms, forward button) and back (back button / swipe), any other directions are just custom animations that could occur with either of the page directions. You can control the animations like the fish demo by appending your own attributes during the events.

Maybe this is not about the fish demo though, in which case I'll need to understand better :)

Reply:
Yes, I would like to have a way to support appending my own attributes :)

Let's look at this example: `... transition:name="main" transition:animate={...} `. 
I could simply write the CSS for a Chrome view transition myself:
```
[data-my-fancy-marker] ::view-transition-new(main) {...} 
```

But I was not able to write the CSS for the VT simulation, e.g. for Firefox, without Astro's help. How can I find the range value `astro-wtmfw64b-2` to write something like the one that is currently generated?
```
[data-my-fancy-marker] [data-astro-transition-fallback="old"][data-astro-transition-scope="astro-wtmfw64b-2"] { ...}
``` 
Is there a way to access the scope identifier from user code? Alternatively I could skip `transition:name` and `transition:animate` and directly set `view-transition-name` on the style-attribute. Then the simulation code could use 
```
[data-my-fancy-marker] [data-astro-transition-fallback="old"][style*="view-transition-name: main"]
```  
That might in fact also be a good way. The idea that the CSS for the real VT and the simulation should be generated from the same definitions might not always lead to good animations.

Reply:
Managed to get it very similar:
```
---
import { ViewTransitions, createAnimationScope } from 'astro:transitions';

const { scope, styles } = createAnimationScope('animationName', { south, north });
---

  <main data-astro-transition-scope={scope}>
  
  <Fragment set:html={styles} />
```

Reply:
Ok, now it works as suggested by Matthew. Thank you very much for this really improving suggestion!